### PR TITLE
Convert spritesheet to TS

### PR DESCRIPTION
**Pull request by eXponenta**
Awaits #6385 because `LoaderResource` type is required

- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/spritesheet/src/SpritesheetLoader.ts [hunk 1]
```diff
@@ -38,14 +40,14 @@ export class SpritesheetLoader
 
         const loadOptions = {
             crossOrigin: resource.crossOrigin,
-            metadata: resource.metadata.imageMetadata,
+            metadata: (resource.metadata as any).imageMetadata,
```
Comment (ivanpopelyshev) [inline packages/spritesheet/src/SpritesheetLoader.ts]: oh no. We cant override metadata anymore! :(

Comment (eXponenta) [reply packages/spritesheet/src/SpritesheetLoader.ts]: All okay while we use v3 of resource loader
https://github.com/pixijs/pixi.js/blob/dev-spritesheet-ts-r/packages/loaders/package.json#L29

Comment (eXponenta) [reply packages/spritesheet/src/SpritesheetLoader.ts]: Then i exposed fields on ResourceLoader class

https://github.com/pixijs/pixi.js/pull/6385/commits/614337808e4329c12e49524cdb770bc4223539a1#diff-aa234d8720b6d5fbb3e622efebb73118R13-R15

File: packages/spritesheet/src/Spritesheet.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import { Rectangle } from '@pixi/math';
-import { Texture } from '@pixi/core';
+import { Texture, BaseTexture, resources } from '@pixi/core';
 import { getResolutionOfUrl } from '@pixi/utils';
 
+type Dict<T> = {[key: string]: T};
```
Comment (bigtimebuddy) [inline packages/spritesheet/src/Spritesheet.ts]: Could you add this to utils in the src/types folder alongside ArrayFixed?

**Comment by ivanpopelyshev** [comment]
Are we adding Dict now? We survived moving core without it

**Comment by bigtimebuddy** [comment]
It’s a helpful utility, it doesn’t hurt anything to add it.

**Pull request merged**